frontend/app/.server/validation/email-address-validation-schema.ts

```diff
@@ -0,0 +1,88 @@
+import isEmail from 'validator/es/lib/isEmail';
+import type { z } from 'zod';
+
+import { createStringValidationSchema } from '~/.server/validation/string-validation-schema';
+
+/**
+ * Interface defining customizable error messages for email address validation schema.
+ */
+export interface EmailAddressValidationSchemaErrorMessages {
+  format?: string;
+  invalidType?: string;
+  maxLength?: string;
+  required?: string;
+}
+
+/**
+ * Configuration options for email address validation, including maximum length and error messages.
+ */
+export interface EmailAddressValidationSchemaOptions {
+  errorMessages?: EmailAddressValidationSchemaErrorMessages;
+  maxLength?: number;
+}
+
+const DEFAULT_MESSAGES = {
+  en: {
+    format: 'Email address format is invalid.',
+    invalidType: 'Email address must be a string.',
+    maxLength: 'Email address must be less than or equal to {{length}} characters.',
+    required: 'Email address is required.',
+  },
+  fr: {
+    format: "Le format de l'adresse courriel est invalide.",
+    invalidType: "L'adresse courriel doit être une chaîne de caractères.",
+    maxLength: "L'adresse courriel doit contenir au maximum {{length}} caractères.",
+    required: "L'adresse courriel est requis.",
+  },
+} as const satisfies Record<Language, Required<EmailAddressValidationSchemaErrorMessages>>;
```

Comment:
The validators should return i18n keys which are resolved at render time by the i18n subsystem.

Reply:
I removed the localization for this pull request for now.